pkg/provisioner/ironic/ironic.go

```diff
@@ -1295,3 +1338,53 @@ func (p *ironicProvisioner) PowerOff() (result provisioner.Result, err error) {
 
 	return result, nil
 }
+
+// softPowerOff sends 'soft power off' request to BM node.
+// If soft power off is not supported, the request ends with an error.
+// Otherwise the request ends with no error and the result should be
+// checked later via node fields "power_state" and "last_error".
+func (p *ironicProvisioner) softPowerOff() (result provisioner.Result, err error) {
+	p.log.Info("ensuring host is powered off by \"soft power off\" command")
+
+	ironicNode, err := p.findExistingHost()
+	if err != nil {
+		return result, errors.Wrap(err, "failed to find existing host")
+	}
+
+	if ironicNode.PowerState != powerOff {
+		// Although request is "soft power off", target state is still "power off"
+		if ironicNode.TargetPowerState == powerOff {
+			p.log.Info("waiting for power status to change")
+			result.RequeueAfter = powerRequeueDelay
+			result.Dirty = true
+			return result, nil
+		}
+		// Checks last error of ironic node, if there is one happened
+		// then soft power off has failed
+		if p.host.Status.SoftPowerOffStatus == metal3v1alpha1.StateSoftPowerOffTriggered {
+			if ironicNode.LastError != "" {
+				err = errors.New(ironicNode.LastError)
+				err = errors.Wrap(err, "soft power off failed")
```

Comment:
I'm not sure this is the sort of thing we should be returning an error on. I think this should just be an ordinary requeue.
This would be a good thing to count in Prometheus though.